test/e2e_node/util.go

```diff
@@ -94,6 +94,19 @@ var (
 	kubeletCfg *kubeletconfig.KubeletConfiguration
 )
 
+func initFeatureGates(featureGates map[string]bool) {
```

Comment:
there is also `serviceFeatureGates`. Can we initialize both without the globals?

Reply:
or rename another method to highlight that it checks kubelet gates, not service gates

Reply:
> there is also serviceFeatureGates. Can we initialize both without the globals?

Yes, looking at it again it seems that `initFeatureGates()` and `SetFeatureGatesForInProcessComponents` basically do the same thing, which is essentially `DefaultMutableFeatureGate.SetFromMap(featureGates)`.

I'm not sure I understand your meaning behind "without the globals". Do you mean that we should have another `MutableFeatureGate` variable (similarly to `DefaultMutableFeatureGate`) that would be dedicated to service gates? Something like:

```go
var (
	DefaultMutableFeatureGate featuregate.MutableFeatureGate = featuregate.NewFeatureGate()
	DefaultMutableServiceGate featuregate.MutableFeatureGate = featuregate.NewFeatureGate()
)
```

Then I can modify `initFeatureGates()` to get the global feature gate variable as an argument, and do something like:
```go
initFeatureGates(featureGates, DefaultMutableFeatureGate)
initFeatureGates(serviceFeatureGates, DefaultMutableServiceGate)
```

We can also create separate methods to check for a feature gate enablement and a service gate enablement.
Was this your meaning? Please let me know if I didn't understand you correctly.

Reply:
Also, I wonder what are service gates exactly and how they differ from "regular" feature gates

Reply:
we are controlling fake API server feature gates separately from kubelet feature gates. See the use of `serviceFeatureGates` variable

Reply:
@SergeyKanzhelev Done, PTAL at the last commit, I hope it aligns with your expectations.

Small question, WDYT about:
https://github.com/kubernetes/kubernetes/blob/7e0a9a7f4e70e4cc6f5dd49917d18101a8e1a25c/test/e2e_node/e2e_node_suite_test.go#L230-L237

Should this code use `serviceFeatureGates` instead of `featureGates`?

Reply:
Unfortunately the linter lane [fails](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/120139/pull-kubernetes-linter-hints/1702230123148742656) since the new `isServiceFeatureGateEnabled()` function is not used =/

What do you suggest? Should we remove it for now?